etna/whatson/models.py

```diff
@@ -0,0 +1,633 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import gettext_lazy as _
+
+from modelcluster.fields import ParentalKey
+from wagtail.admin.panels import FieldPanel, InlinePanel, MultiFieldPanel
+from wagtail.fields import RichTextField
+from wagtail.images import get_image_model_string
+from wagtail.models import Orderable
+from wagtail.search import index
+from wagtail.snippets.models import register_snippet
+
+from etna.articles.models import ArticleTagMixin
+from etna.collections.models import TopicalPageMixin
+from etna.core.models import BasePageWithIntro
+
+
+class VenueType(models.TextChoices):
+    """
+    This model is used to add venue types to event pages.
+    """
+
+    ONLINE = "online", _("Online")
+    IN_PERSON = "in_person", _("In person")
+    HYBRID = "hybrid", _("Hybrid")
+
+
+@register_snippet
+class EventType(models.Model):
+    """
+    This snippet model is used so that editors can add event types,
+    which we use via the event_type ForeignKey to add event types
+    to event pages.
+    """
```

Comment:
@jamesbiggs How many options are there? Does this need to be a model, or would `TextChoices` do here as well? Another good reason for using `TextChoices` is when you have bits of conditional code based around the values.

Reply:
If we DO indeed need a model here, we should add a `slug` field for use in public URLs

Reply:
I asked about having hard-coded EventTypes or having a snippet so that editors can update/create new ones.

Currently it's:
Event
Course
Workshop
Training
Talk
Conference
Tour
Professional development

I shall add a slug! 😄